website/docs/r/security_group.html.markdown

```diff
@@ -96,6 +96,24 @@ resource "aws_vpc_endpoint" "my_endpoint" {
 
 You can also find a specific Prefix List using the `aws_prefix_list` data source.
 
+### Change of name or name-prefix value
+
+The `name` and `name-prefix` arguments force the creation of a new Security Group resource when they change value. In that case, Terraform first deletes the existing Security Group resource and then it creates a new one. If the existing Security Group is associated to a Network Interface resource, the deletion cannot complete. The reason is that Network Interface resources cannot be left with no Security Group attached and the new one is not yet available at that point.
```

Comment:
If you can link to AWS public documentation to cite this, that would be a welcomed addition (but not required).